dart/lib/src/transport/transport.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+import 'package:sentry/src/utils.dart';
+
+import '../protocol.dart';
+import '../sentry_options.dart';
+import 'body_encoder_browser.dart' if (dart.library.io) 'body_encoder.dart';
+import 'header_builder_browser.dart' if (dart.library.io) 'header_builder.dart';
+
+typedef BodyEncoder = List<int> Function(
+  Map<String, dynamic> data,
+  Map<String, String> headers, {
+  bool compressPayload,
+});
+
+/// A transport is in charge of sending the event to the Sentry server.
+class Transport {
+  final SentryOptions _options;
+
+  @visibleForTesting
+  final Dsn dsn;
+
+  /// Use for browser stacktrace
+  final String origin;
+
+  /// Used by sentry to differentiate browser from io environment
+  final String platform;
+
+  final Sdk sdk;
+
+  CredentialBuilder _credentialBuilder;
+
+  Transport({
+    @required SentryOptions options,
+    @required this.sdk,
+    @required this.platform,
+    this.origin,
+  })  : _options = options,
+        dsn = Dsn.parse(options.dsn) {
+    _credentialBuilder = CredentialBuilder(
+      dsn: Dsn.parse(options.dsn),
+      clientId: sdk.identifier,
+    );
+  }
+
+  Future<SentryId> send(SentryEvent event) async {
+    final now = _options.clock();
+
+    var authHeader = _credentialBuilder.build(now.millisecondsSinceEpoch);
+    final headers = buildHeaders(authHeader, sdk: sdk);
```

Comment:
this could be assigned to a private global field on the ctor body, so we don't need to do on every event